pkg/maps/maps.go

```diff
@@ -39,3 +39,11 @@ func Values[M ~map[K]V, K comparable, V any](m M) []V {
 func Keys[M ~map[K]V, K comparable, V any](m M) []K {
 	return maps.Keys(m)
 }
+
+// CopyRightIntoLeft copies all key/value pairs in src adding them to dst.
+// When a key in src is already present in dst,
+// the value in dst will be overwritten by the value associated
+// with the key in src.
+func CopyRightIntoLeft[M1 ~map[K]V, M2 ~map[K]V, K comparable, V any](dst M1, src M2) {
```

Comment:
nit: why not named `Copy` or `Merge` ?

Reply:
https://github.com/istio/istio/pull/45510#discussion_r1235596688
It is discussed, I prefer to use Copy.